releng_sop/koji_sign.py

```diff
@@ -0,0 +1,715 @@
+            paths.append(dst_path)
+
+        return temp_dir, paths
+
+    def sign_rpms_in_temp(self, sigkey, paths, commit=False):
+        """
+        Sign RPMs in temp (copies!) with a sigkey.
+
+        The signing procedure is determined by self.rpmsign_class instance.
+
+        :param sigkey: Sigkey
+        :type  sigkey: str
+        :param paths: Paths to RPM packages
+        :type  paths: list
+        :param commit: Disable dry-run, apply changes for real.
+        :type  commit: bool=False
+        """
+        if not commit:
+            return
+        sign = self.rpmsign_class()
+        sign.sign(sigkey, paths)
+
+    def clean_temp(self, temp_dir, paths, commit=False):
+        """
+        Remove signed RPMs from temp, then remove the temp dir.
+
+        :param temp_dir: Path to a temp directory with signer RPMs
+        :type  temp_dir: str
+        :param paths: Paths to RPM packages
+        :type  paths: list
+        :param commit: Disable dry-run, apply changes for real.
+        :type  commit: bool=False
+        """
+        if not commit:
+            return
+        for path in paths:
+            os.remove(path)
+        os.rmdir(temp_dir)
+
+    def import_signed_rpms(self, rpm_info_list, paths, sigkey, rpm_sig_dict=None, commit=False):
+        """
+        Import signed RPMs from temp to koji.
+
+        :param rpm_info_list: List of koji rpm_info dictionaries
+        :type  rpm_info_list: list
+        :param paths: Paths to RPM packages
+        :type  paths: list
+        :param sigkey: Sigkey
+        :type  sigkey: str
+        :param rpm_sig_dict: A dictionary obtained from get_rpm_sig_dict() method
+        :type  rpm_sig_dict: dict
+        :param commit: Disable dry-run, apply changes for real.
+        :type  commit: bool=False
+        """
+        if not commit:
+            return
+
+        # check if lists are equally long
+        if len(rpm_info_list) != len(paths):
+            raise ValueError("Arguments 'rpm_info_list' and 'paths' must be equally long")
+
+        # check if filenames match at each position
+        for rpm_info, path in zip(rpm_info_list, paths):
+            rpm_info_fn = os.path.basename(self._get_rpm_path(rpm_info, None))
+            path_fn = os.path.basename(path)
+            if rpm_info_fn != path_fn:
+                raise ValueError("File names in 'rpm_info' and 'path' do not match: %s vs %s" % (rpm_info_fn, path_fn))
+
+        if not rpm_sig_dict:
+            rpm_sig_dict = self.get_rpm_sig_dict(rpm_info_list)
+
+        for (rpm_info, path) in zip(rpm_info_list, paths):
+            rpm_sighdr, rpm_sigkey = self._get_rpm_sighdr_sigkey(path)
+            if rpm_sigkey != sigkey:
+                raise ValueError("Expected sigkey: %s; RPM is signed with '%s': %s" % (sigkey, rpm_sigkey, path))
+            rpm_sighdr_base64 = base64.encodestring(rpm_sighdr)
+            self.koji_session.addRPMSig(rpm_info["id"], rpm_sighdr_base64)
```

Comment:
I'm thinking if use of try except could be beneficial here?
Looking into koji source code I see that the ``addRPMSig()`` method may raise ``koji.GenericError`` even for situation when the signature already exist:

    raise koji.GenericError, "Signature already exists for package %s, key %s" % (nvra, sigkey)

IMHO, this kind of error could be ignored and warning could be sufficient. If multiple signing tasks is running for different products but these products have some packages in common, then this error may happen, although it's not fatal and there is nothing the caller can do except to run it again.